ocean_provider/validation/algo.py

```diff
@@ -283,68 +283,78 @@ def validate(self):
 
         return self.validate_usage()
 
-    def validate_algo(self):
-        """Validates algorithm details that allow the algo dict to be built."""
-        algorithm_meta = self.data.get("algorithmMeta")
-        algorithm_did = self.data.get("algorithmDid")
-
-        privacy_options = self.service.main.get("privacy", {})
-
-        if algorithm_meta and privacy_options.get("allowRawAlgorithm", True) is False:
-            self.error = f"cannot run raw algorithm on this did {self.did}."
-            return False
-
-        trusted_algorithms = privacy_options.get("publisherTrustedAlgorithms", [])
-
-        if not (algorithm_did and trusted_algorithms):
-            return True
-
+    def _validate_trusted_algos(self, algorithm_did, trusted_algorithms):
         try:
-            trusted_dids = [algo["did"] for algo in trusted_algorithms]
-            if algorithm_did not in trusted_dids:
+            did_to_trusted_algo_dict = {algo["did"]: algo for algo in trusted_algorithms}
+            if algorithm_did not in did_to_trusted_algo_dict:
                 self.error = f"this algorithm did {algorithm_did} is not trusted."
-                return False
+
+            elif (
+                    not did_to_trusted_algo_dict[algorithm_did].get("filesChecksum") or
+                    not did_to_trusted_algo_dict[algorithm_did].get("containerSectionChecksum")
+            ):
+                self.error = "The trusted algorithm dict is not valid, missing " \
+                             "filesChecksum and/or containerSectionChecksum."
+
         except KeyError:
+            did_to_trusted_algo_dict = {}
             self.error = (
                 "Some algos in the publisherTrustedAlgorithms don't have a did."
             )
-            return False
 
-        for trusted_algorithm in trusted_algorithms:
-            algo = get_asset_from_metadatastore(
-                get_metadata_url(), trusted_algorithm["did"]
+        if not self.error:
```

Comment:
no need for this if. whenever self.error is set, return False. Then you can remove the if and continue the flow of testing with its contents